### PR TITLE
Backend/feat: send dummy notification to driver via dashboard

### PR DESCRIPTION
**Pull request by jaypalmudaliyar24**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added a new dashboard API to send dummy notification to driver.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Motivation and Context

Added this API to test FCM channel via dashboard.

## How did you test it?

Tested locally, was able to get logs for `mock-fcm` getting used.

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Comment by 0utkarsh** [comment]
Can you mention why this feature is needed?

**Comment by jaypalmudaliyar24** [comment]
> Can you mention why this feature is needed?
It was needed to test FCM channel via dashboard.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs [hunk 1]
```diff
@@ -1586,3 +1590,94 @@ getLimitAccordingToChannel config channel =
...
+
+  let personId = cast @Common.Driver @DP.Person driverId
+  driver <- B.runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
+  -- merchant access check
+  unless (merchantOperatingCity.id == driver.merchantOperatingCityId) $ throwError (PersonDoesNotExist personId.getId)
+
+  now <- getCurrentTime
+  let deviceToken = driver.deviceToken
+      entityData = mkDummyNotificationEntityData now
+  _ <- TN.notifyOnNewSearchRequestAvailable merchantOperatingCity.id driver.id deviceToken entityData
```
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs]: Fixed.

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs [hunk 2]
```diff
@@ -1586,3 +1590,94 @@ getLimitAccordingToChannel config channel =
...
+  pure Success
+
+dummyId :: Text
+dummyId = ""
+
+mkDummyNotificationEntityData :: UTCTime -> DSearchReq.SearchRequestForDriverAPIEntity
+mkDummyNotificationEntityData now =
+  let thirtySeconds = 30
+      searchRequestValidTill = addUTCTime thirtySeconds now
```
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs]: Fixed.

File: Backend/dev/migrations/provider-dashboard/0028-make-server-array-merchant.sql [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ALTER TABLE atlas_bpp_dashboard.merchant ADD COLUMN server_names Text[];
 
 UPDATE atlas_bpp_dashboard.merchant as T1
-SET server_names = ARRAY[server_name];
+SET server_names = ARRAY[server_name, 'DRIVER_OFFER_BPP_MANAGEMENT'];
```
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/dev/migrations/provider-dashboard/0028-make-server-array-merchant.sql:4]: Addee new migrations.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs [hunk 4]
```diff
@@ -1602,3 +1606,131 @@ getLimitAccordingToChannel config channel =
     ALERT -> config.alert
 
 --------------------------------------------------------------------------------------------------
+
+--------------------------------------------------------------------------------------------------
+
+sendDummyNotificationToDriver :: ShortId DM.Merchant -> Context.City -> Id Common.Driver -> Flow APISuccess
```
[the remark this answers is not in the document]
Comment (jaypalmudaliyar24) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs:1612]: Should i create a new module `Domain.Action.Dashboard.Driver.Notification` and move this function there?

**Review by hkmangla — APPROVED** [review]

**Pull request merged**